### ISSUE TITLE
Allow lists of settings without unfolding

### ISSUE THREAD

**Issue by johanandren** [sbt/sbt#1682]
A lot of plugins (and local project configuration) provides a list of settings, when adding those to a Project those lists needs to be unfolded into the settings method, like this example with the sbt-native-packager:

``` scala
lazy val bananas = Project("bananas-app", file("app"))
  .settings(packageArchetype.java_application: _*)
```

When doing a sbt workshop with the Scala usergroup here in Stockholm, this was something a lot of people stumbled upon, should be pretty non-invasive to add a settings that takes a IterableOnce or something like that and allow us to call it like this instead:

``` scala
lazy val bananas = Project("bananas-app", file("app"))
  .settings(packageArchetype.java_application)
```

**Comment by johanandren** [comment]
Sorry, missed the part in CONTRIBUTING about asking on sbt-dev first.

**Comment by jsuereth** [comment]
Yes. I'm going to close for a discussion in sbt-dev.

Basically, the reason `settings` HAS to be the varargs i because you can't overload with Seq[Setting[_]] thanks to how scala encodes varags.

However, an alternative would be ok, IMHO.   Let's discuss on sbt-dev.

**Comment by dwijnand** [comment]
fixed with https://github.com/sbt/sbt/pull/1902 in 0.13.8

also:

> the reason settings HAS to be the varargs i because you can't overload with Seq[Setting[_]] thanks to how scala encodes varags.

you can overload by defining `def foo(x0: A, xs: A*) = foo(x0 +: xs)` (but that's not how I fixed it back in 0.13.8).
